### ISSUE TITLE
Disabling powerline breaks diminish minor mode functionality

### ISSUE THREAD

**Issue by sooheon** [syl20bnr/spacemacs#2663]
This is a recent regression, where if `powerline` is excluded, minor modes are not properly diminished to the single letter toggle keys. To reproduce, simple add `powerline` to dotspacemacs-excluded-packages.

**Comment by sooheon** [comment]
I can see this is because `spacemacs//mode-line-prepare` is within powerline use-package init. Specifically, if the following L3191~3201 is eval'ed separately, I get the diminished modes in the default mode-line. Maybe it's as simple as factoring this out to a different function which runs regardless of powerline? If you tell me where you want it, @syl20bnr, I could make a PR.
```
(let ((unicodep (dotspacemacs|symbol-value
                           dotspacemacs-mode-line-unicode-symbols)))
            (dolist (mm spacemacs--diminished-minor-modes)
              (let ((mode (car mm)))
                (when (and (boundp mode) (symbol-value mode))
                  (let* ((unicode (cadr mm))
                         (ascii (caddr mm))
                         (dim (if unicodep
                                  unicode
                                (if ascii ascii unicode))))
                    (diminish mode dim))))))
```

**Comment by syl20bnr** [comment]
Oops I see this may be tricky get it right outside the powerline, or maybe not.

You are right, the first thing is to extract the code to a function. Then ideally the function should be called in `prepare` of powerline, and in the case of no powerline, it should be called somewhere in the mode-line format. This is important because this shenanigan allow to support the use case mentioned in this commit: https://github.com/syl20bnr/spacemacs/commit/e410812d59c0220c151498a183766bb6331dccb3

The easy solution but not ideal is to call this function at least once somewhere to initialize the diminished lighters. But you won't have the benefits mentioned in the commit above.

**Comment by syl20bnr** [comment]
Should be fixed in develop.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
